include/zmq.h

```diff
@@ -371,6 +374,18 @@ ZMQ_EXPORT const char *zmq_msg_gets (const zmq_msg_t *msg, const char *property)
 #define ZMQ_VMCI_BUFFER_MAX_SIZE 87
 #define ZMQ_VMCI_CONNECT_TIMEOUT 88
 #define ZMQ_USE_FD 89
+#define ZMQ_PEER_EVENTS 90
+
+/*  getsockopt additional struct to recover peer events                       */
+typedef struct zmq_gso_peer_events_t
+{
+    void *zid;
+    int zid_len;
+    short events;
+    short padding;
```

Comment:
What's the purpose of the padding?

Reply:
Align to 4 byte multiple and save space for any eventual addition (binary compatibility). But it can be removed.

Reply:
I think you are making several assumptions on the size of the members and on the alignment that may not be true on all platforms.

Reserving something for addition without breaking binary compatibility may still be a good idea, but then I would rename the member to "reserved".